velox/functions/sparksql/String.h

```diff
@@ -583,4 +583,167 @@ struct OverlayVarbinaryFunction : public OverlayFunctionBase {
   }
 };
 
+/// translate(string, match, replace) -> varchar
+///
+///   Returns a new translated string. It translates the character in ``string``
+///   by a character in ``replace``. The character in ``replace`` is
+///   corresponding to the character in ``match``. The translation will
+///   happen when any character in ``string`` matching with a character in
+///   ``match``.
+template <typename T>
+struct TranslateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  folly::F14FastMap<std::string, std::string> dictionaryUtf8_;
+  folly::F14FastMap<char, char> dictionaryAscii_;
+
+  bool isConstantTranslateDictionary = false;
+  bool isConstantTranslateDictionaryBuilt = false;
```

Comment:
This is not needed. Use isAscii function from ./velox/functions/lib/string/StringCore.h

```
/// Check if a given string is ascii
static bool isAscii(const char* str, size_t length);
```

Reply:
Hi @mbasmanova, `isConstantTranslateDictionaryBuilt` is used for marking whether the dictionary is built for constant arg case. Since it it impossible to identify all-ascii input in `initialize()`, we uniformly build in the corresponding kind of dictionary call/callAscii. Am I missing something? Thanks!